modules/iopcore/common/cdvd_config.h

```diff
@@ -17,7 +17,9 @@ struct cdvdman_settings_common
     u16 flags;
     u32 layer1_start;
     u8 DiscID[5];
-    u8 padding[3];
+    u8 bdm_cache;
+    u8 hdd_cache;
+    u8 smb_cache;
```

Comment:
We can never have 3 zso caches at the same time, so we do not need 3 settings to the iopcore. Changing these 3 into 1 `zso_cache` will also remove the need to have an extra function defined in the device layer to get the size of the zso cache.

Please note that in the GUI it does make sense to have 3 separate settings. Only when starting 1 of the 3 device types, we can pass 1 of the 3 settings to the iopcore.

Reply:
Very good idea!
This would allow us to also have more configuration options (like enabling/disabling the cache for ISO too).